src/peft/tuners/lora/variants.py

```diff
@@ -130,6 +130,186 @@ def forward(module: Linear, active_adapter: str, x: torch.Tensor, result: torch.
         return result
 
 
+class QALoraLinearVariant(LoraVariant):
+    @staticmethod
+    def init(module: Linear, adapter_name: str, **kwargs: Any) -> None:
+        """
+        Initializes QALoRA specific parameters for a given adapter.
+
+        Args:
+            module (Linear): The linear module to be adapted.
+            adapter_name (str): The name of the adapter.
+            **kwargs: Additional keyword arguments.
+                qalora_group_size (int, optional): The size of groups for pooling. Defaults to 16.
+        """
+        # Store the qalora_group_size as a module attribute
+        # It's expected that qalora_group_size is passed in kwargs from LoraConfig
+        qalora_group_size = kwargs.get("qalora_group_size", 16)  # Default to 16 if not specified in kwargs
+        if not hasattr(module, "qalora_group_size"):
+            module.qalora_group_size = {}
```

Comment:
We need to add: `module.other_param_names = module.other_param_names[:] + ("qalora_group_size",)`

This corresponds to this DoRA code:

https://github.com/huggingface/peft/blob/e67052b18cb3c58370e0339ecf76a5faf447f2ff/src/peft/tuners/lora/variants.py#L33